quinn-proto/src/connection.rs

```diff
@@ -1842,7 +1844,7 @@ impl Connection {
                 size: buf.len() as u16,
                 is_crypto_packet: crypto_level == CryptoLevel::Initial && !ack_only,
                 ack_eliciting: !ack_only,
-                in_flight: !ack_only,
+                in_flight: padded || !ack_only,
```

Comment:
This has no effect as-is, but will become more relevant when e.g. bd91d282cf4404090ccd6331b933f8d5686a671e is merged.